File: Leetcode/Python/Finished/Easy/ReverseInt.py
def reverse(x: int):
    """
    reverses the given number
    :param x: int
    :return: int
    """
    if x < 0:
        signed = True
    else:
        signed = False
    x = str(x)
    if signed:
        x = x[1:]
    x = x[::-1]
    x = int(x)
    if signed:
        x = -x
    if x > 2 ** 31 - 1:
        return 0
    if x < -2 ** 31:
        return 0
    else:
        return x


def test_reverse():
    assert reverse(-123) == -321
    assert reverse(123) == 321
    assert reverse(123456789) == 987654321
    assert reverse(100000) == 1
    assert reverse(-21448484858680600) == 0
    assert reverse(203340304303030) == 0
    assert reverse(2**31) == 0
    assert reverse(-2**31 -1) == 0

if __name__ == "__main__":
    test_reverse()
    print("All Tests Passed")

File: Leetcode/JavaScript/Finished/Easy/reverse_int.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    let signed;
    if (x < 0) {
        signed = true;
    }
    else {
        signed = false;
    }
    x = x.toString();
    if (signed) {
        x = x.substring(1);
    }
    x = x.split('');
    let i;
    let answer = "";
    for (i = x.length - 1; i >= 0; i--) {
        answer += x[i];
    }
    answer = parseInt(answer);
    if (signed) {
        answer = -answer;
    }
    if (answer > Math.pow(2, 31) - 1) {
        return 0
    }
    if (answer < Math.pow(-2, 31)) {
        return 0
    }
    return answer;
};